### ISSUE TITLE
RNN notebook output does not match

### ISSUE THREAD

**Issue by ashuarsh** [stefan-jansen/machine-learning-for-trading#127]
code block [30] in https://github.com/stefan-jansen/machine-learning-for-trading/blob/master/19_recurrent_neural_nets/01_univariate_time_series_regression.ipynb

seems to be only trained on 96 points?

Epoch 1/150
95/**96** [============================>.] - ETA: 0s - loss: 0.0162
Epoch 00001: val_loss improved from inf to 0.00766, saving model to results/univariate_time_series/rnn.h5

This does not match the output when one runs on their own 

Epoch 1/150
1900/**1914** [============================>.] - ETA: 0s - loss: 0.1076
Epoch 00001: val_loss improved from inf to 0.02894, saving model to results/univariate_time_series/rnn.h5

See block [23] which does say the y_train.shape is 1914

Can you confirm this? Or perhaps I'm missing something silly?

**Comment by stefan-jansen** [comment]
The stats are per batch, each of which includes 20 samples (see parameter `batch_size`; 1914/20=95.7 so you get around 95 or 96 iterations per epoch. Hope this helps!

**Comment by ashuarsh** [comment]
Thanks so much for your prompt reply Stefan 

Also the plot in block [31] seems incorrect: the best epoch (black dotted line) should intersect the validation RMSE curve at it's minima - sorry if I'm missing something

I would love to reproduce the stats in the notebook - would you know which parameter in https://www.tensorflow.org/api_docs/python/tf/keras/Sequential switches the stats to be printed per batch?

**Comment by stefan-jansen** [comment]
The plot shows the moving average since the loss is somewhat volatile esp visually on log scale whereas the vertical line does indeed highlight the minimum (see use of `.idxmin()`). 

When I said 'stats' above I was referring to how the progress is shown as this is necessarily printed by batch since this is how the data is processed. `batch_size` sets the size of the batch. What do you mean by reproducing the stats?

**Comment by ashuarsh** [comment]
Thanks for the clarification - my apologies for not reading through carefully - the plot for me does intersect the minima (probably my loss isn't as volatile)

Btw this is a beautiful body of work - congratulations! And very generous!

As some one who's only dipping their toes into it, it's encouraging to be able to reproduce the stats you show in the notebook - lets me know I haven't done something silly / gone astray - so I was just saying that running the same notebook doesn't print the stats by batch, instead:

Train on 1914 samples, validate on 252 samples
Epoch 1/150
1900/1914 [============================>.] - ETA: 0s - loss: 9.2338e-05
Epoch 00001: val_loss did not improve from 0.00011
1914/1914 [==============================] - 16s 8ms/sample - loss: 9.2913e-05 - val_loss: 6.2522e-04
Epoch 2/150
....

I suppose I should take this as evidence I must've done something wrong.

**Comment by stefan-jansen** [comment]
Not wrong, you probably just changed some setting, e.g. `batchsize=1`. You definitely should experiment with different configurations, I'm not claiming that the settings used in the notebooks perform best, they should just reflect reasonable choices.
